### PR TITLE
HN 002 : Add See Comments

### PR DESCRIPTION
**Pull request by AsminBudha**
1. Add single post when click to post.
2. Add comments and sub comments.

### CODE REVIEW

File: src/AppRouter.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
+
+import App from './App';
+import Header from './components/Header';
+import NotFound from './components/NotFound';
+import AppConstants from './constants/common';
```
Comment (badyakarprabin) [inline src/AppRouter.js]: ``` import * as appConstants from './constants/common'; ```

File: src/AppRouter.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+   * Renders JSX component.
+   *
+   * @returns
+   * @memberof AppRouter
+   */
+  render() {
+
+    return (
+      <Router>
+        <div>
```
Comment (badyakarprabin) [inline src/AppRouter.js:28]: Why do we need div here?

File: src/components/Comments.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+
+import SingleComment from './SingleComment';
+
+/**
+ * Component for comments box and comments list.
+ *
+ * @param {Object} props Object of params send by calling object.
+ */
+const Comments = (props) => {
```
Comment (badyakarprabin) [inline src/components/Comments.js]:
``` 
const Comments = ({data = [] }) => {

  const items = data
    .map((id) => {
      return <li key={id}><SingleComment id={id} /></li>;
    });
```

No need to use const kids.

File: src/components/Header.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+
+/**
+ * App title .
+ */
+const Header = () => <h1>Hacker News</h1>;
```
Comment (badyakarprabin) [inline src/components/Header.js]: ``` <h1> {appContants.HEADER} </h1> ```

You might have used Hacker News in multiple places. What if someone asked you to change your application name form Hacker New to Hacker Post ?

File: src/components/NewsItem.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  /**
+   * This function is called automatically after render is finished.
+   * It retrieve data of current item from API.
+   *
+   * @memberof NewsItem
+   */
+  componentDidMount() {
+    const { id } = this.props;
+
+    Http
```
Comment (badyakarprabin) [inline src/components/NewsItem.js]: ``` http.get ```

File: src/components/NewsItem.js [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+   * @returns
+   * @memberof NewsItem
+   */
+  render() {
+    const { data } = this.state;
+    const element = data ?
+      <div>
+        <h4>{data.title}</h4>
+        <div>
+
```
Comment (badyakarprabin) [inline src/components/NewsItem.js:65]: Extra line

File: src/components/NewsItem.js [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+      });
+  }
+
+  /**
+   *
+   * @returns
+   * @memberof NewsItem
+   */
+  render() {
+    const { data } = this.state;
```
Comment (badyakarprabin) [inline src/components/NewsItem.js:63]: ``` const {data: {title, by, time}} ```

File: src/components/NewsItem.js [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+    const element = data ?
+      <div>
+        <h4>{data.title}</h4>
+        <div>
+
+          author : <span>{data.by} </span>
+          created At : <span>{new Date(data.time).toLocaleString()} </span>
+          type : <span>{data.type}</span>
+          <span onClick={() => window.location.href = data.url}> Read More</span>
```
Comment (badyakarprabin) [inline src/components/NewsItem.js]: ``` onClick={this.onChangeLocation} ```

File: src/components/NewsItem.js [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+          Comments:
+          <Link to={
+            {
+              pathname: `${AppConstants.API_ITEM}/:${data.id}`,
```
Comment (badyakarprabin) [inline src/components/NewsItem.js]: .

File: src/components/NewsList.js [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+    });
+  }
+
+  /**
+   * Update page number with value by either increase or decrease.
+   *
+   * @param {number} by Increase or Decrease value normally +-1.
+   * @memberof NewsList
+   */
+  updatePage = (by) => {
```
Comment (badyakarprabin) [inline src/components/NewsList.js]: What is meaning of by here???

File: src/components/NewsList.js [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+
+  /**
+   * This function is automatically called after render is done.
+   * Retrieve news ids from API and set into state.
+   *
+   * @memberof App
+   */
+  componentDidMount() {
+    const path = this.checkPath();
+    const dataIds = Http.get(path);
```
Comment (badyakarprabin) [inline src/components/NewsList.js]: .

File: src/components/NewsList.js [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+          disabled={prevBtnDisabledStatus}
+          onClick={() => this.updatePage(AppConstants.PAGINATION_DECREMENT_FACTOR)}
+        >
+          Prev
+        </button>
+        <button
+          disabled={nextBtnDisabledStatus}
+          onClick={() => this.updatePage(AppConstants.PAGINATION_INCREMENT_FACTOR)}
+        >
+          Next
```
Comment (badyakarprabin) [inline src/components/NewsList.js]: ``` appConstants.NEXT```

File: src/components/NotFound.js [hunk 13]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+/**
+ * Function which renders a text Page Not Found.
+ *
+ */
+const notFound = () => <h1>404 Page Not Found</h1>;
+
+export default notFound;
```
Comment (badyakarprabin) [inline src/components/NotFound.js]: Component name always starts with capital letter.
``` 
const NotFound = () ....

export default NotFound 
```

File: src/components/SingleComment.js [hunk 14]
```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+
+import Comment from './Comments';
+import Http from '../services/http';
```
Comment (badyakarprabin) [inline src/components/SingleComment.js]: Only component starts with capital letter.

File: src/components/SingleComment.js [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+import AppConstants from '../constants/common';
```
Comment (badyakarprabin) [inline src/components/SingleComment.js]: same

File: src/components/SingleComment.js [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+   * Creates an instance of SingleComment.
+   *
+   * @param {*} props
+   * @memberof SingleComment
+   */
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      data: null,
```
Comment (badyakarprabin) [inline src/components/SingleComment.js:23]: null or []?

File: src/components/SinglePost.js [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+ * It shows post title and comments.
+ *
+ * @param {Object} props
+ */
+const SinglePost = (props) => {
+  const { data } = props;
+
+  return (
+    <div>
+      <h2>{data.title}</h2>
```
Comment (badyakarprabin) [inline src/components/SinglePost.js]: yesma k garna milxa ho????

File: src/components/Tabs.js [hunk 18]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * Component for tabs in app.
+ *
+ */
+const Tabs = () => {
+  return (
+    <div className="nav-content red">
+      <ul className="tabs tabs-transparent">
+        <li className="tab"><Link to={Common.API_TOP_STORIES} className='active'>Top Stories</Link></li>
```
Comment (badyakarprabin) [inline src/components/Tabs.js]: Top stories shoud be imported from constants files

File: src/constants/common.js [hunk 19]
```diff
@@ -0,0 +1,27 @@
+const TOP_STORIES = 0;
```
Comment (badyakarprabin) [inline src/constants/common.js]: yo gare ni hunxa or
```
export const NEW_STORIES = 1;
export cont BEST_STORIES = 2;

import * as appConstants from 'common';
```

**Review by badyakarprabin — CHANGES_REQUESTED** [review]

File: src/AppRouter.js (hunk 2 above)
Comment (AsminBudha) [reply src/AppRouter.js:28]: Router should have single child.

File: src/components/SingleComment.js (hunk 16 above)
Comment (AsminBudha) [reply src/components/SingleComment.js:23]: data store single object not an array.

**Pull request merged**